k2/python/k2/autograd.py

```diff
@@ -322,11 +322,14 @@ def forward(ctx, src: torch.Tensor, index: torch.Tensor) -> torch.Tensor:
     def backward(ctx, out_grad) -> Tuple[torch.Tensor, None]:
         src, index = ctx.saved_tensors
 
-        ans = torch.zeros(src.size(0),
+        ans = torch.zeros(src.shape,
                           dtype=torch.float32,
                           device=src.device,
                           requires_grad=False)
-        _k2.index_add(index, out_grad, ans)
+        if out_grad.stride()[0] == 0:
+            # TODO(fangjun): why is the stride of out_grad 0 ??
```

Comment:
I tried to figure out why `out_grad.stride() == (0, 0)` but failed.
@danpovey @qindazhu could you give some comments about it?

Reply:
have you ever printed some logs at C++ level of `index_select`?

Reply:
Yes, I have a check inside c++ to ensure that the stride for the column is 1. But the actual value is 0. That is
how I discovered that the `out_grad.stride() == (0, 0)`.

`out_grad` is generated by autograd from Python.

Reply:
What is the dimension of out_grad?

Reply:
> I have a check inside c++ to ensure that the stride for the column is 1. But the actual value is 0

I suppose you add a check locally(sorry, I didn't find it with current code), so you mean even in C++, the stride is 0? when we create tensor with something like `src.options()`? what if we add `stride` manually?

Reply:
> I suppose you add a check locally(sorry, I didn't find it with current code),

Here is the check in the code:

https://github.com/k2-fsa/k2/blob/da3c30e0d277a69013610bf0838592f25e003861/k2/python/csrc/torch/index_add.cu#L167

where `value` is actually `out_grad` from Python:
https://github.com/k2-fsa/k2/blob/da3c30e0d277a69013610bf0838592f25e003861/k2/python/k2/autograd.py#L329-L332

Reply:
fixed.